### ISSUE TITLE
CURL data leaking when expanding the latent

### ISSUE THREAD

**Issue by Dovermore** [deepmind/deepmind-research#139]
Hi all,
Is the following section of code a bug leaking data from training set to unsupervised learning when the model is being expanded? It seems to use directly the label instead of the paper proposed `new component`. At least from logging, I see the value of `label_onehot_batch` directly corresponds to the actual label
```
            for _ in range(num_buffer_train_steps):
              for bs in range(n_poor_batches):
                x_batch = poor_data_buffer[bs * batch_size:(bs + 1) *
                                           batch_size]
                label_batch = poor_data_labels[bs * batch_size:(bs + 1) *
                                               batch_size]
                label_onehot_batch = np.eye(n_y)[label_batch]        # -----> should be np.eye(n_y).matmul(np.array([new_cluster]*bs))?
                _ = sess.run(
                    train_step_supervised,
                    feed_dict={
                        x_train_raw: x_batch,
                        model_train.y_label: label_onehot_batch
                    })
```

**Comment by drao2** [comment]
Thanks for spotting this and for reaching out! After taking a look, it does appear you're correct, and the fine-tuning steps on expansion are using the class labels. Thankfully, this doesn't seem to be a situation which benefits the model, since the cluster dimensions don't correspond to classes, and so it is effectively using "incorrect labels" for a few steps.

We've run some initial tests, and the results don't seem to change (within the error bounds), but we'll run the larger set of experiments to make sure, and update the paper as necessary.

I'll also submit a fix in the next few days. Thanks again for raising this.

**Comment by drao2** [comment]
Sorry for the delay on this, and thanks for your patience. I've run a much larger set of experiments from the paper, and indeed, this doesn't really affect the results.

I've submitted a fix that does correct training on expansion, it should land in the next sync.

Thanks again for raising this!

**Comment by drao2** [comment]
Sorry for the delay on this, and thanks for your patience. I've run a much larger set of experiments from the paper, and indeed, this doesn't have much of an impact on the results.

I've submitted a fix that does correct training on expansion, it should land in the next sync.

Thanks again for raising this!
